pybikes/velobike_ru.py

```diff
@@ -26,30 +26,49 @@ def update(self, scraper=None):
         stations = []
 
         data = json.loads(scraper.request(self.feed_url))
-        # Each station is
+
+        #Each station is
         #{
-        #  "Address": "415 - \u0443\u043b. \u0421\u0443\u0449\u0451\u0432\u0441\u043a\u0438\u0439 \u0412\u0430\u043b, \u0434.2",
-        #  "FreePlaces": 12,
-        #  "Id": "0415",
-        #  "IsLocked": true,
+        #  "StationTypes":[
+        #    "electric",
+        #    "ordinary"
+        #  ],
+        #  "TotalOrdinaryPlaces":42,
+        #  "IconsSet":2,
+        #  "AvailableOrdinaryBikes":48,
+        #  "FreeElectricPlaces":1,
+        #  "FreeOrdinaryPlaces":5,
+        #  "IsFavourite":false,
+        #  "TotalPlaces":66,
+        #  "HasTerminal":false,
         #  "Position": {
-        #    "Lat": 55.7914268,
-        #    "Lon": 37.5905396
+        #    "Lat":55.7351049,
+        #    "Lon":37.5937289
         # },
-        # "TotalPlaces": 12
+        #  "Address":"\u0417\u0443\u0431\u043e\u0432\u0441\u043a\u0438\u0439 \u0431-\u0440, \u0434.5, \u0441\u0442\u0440.1 (\u0441\u0442. \u043c. \u041f\u0430\u0440\u043a \u041a\u0443\u043b\u044c\u0442\u0443\u0440\u044b)",
+        #  "AvailableElectricBikes":16,
+        #  "Id":"0100",
+        #  "TotalElectricPlaces":24,
+        #  "TemplateId":6,
+        #  "IsLocked":true,
+        #  "FreePlaces":6,
+        #  "Name":""
         #}
         for item in data['Items']:
             name = item['Address']
             latitude = float(item['Position']['Lat'])
             longitude = float(item['Position']['Lon'])
-            bikes = int(item['TotalPlaces'])-int(item['FreePlaces'])
+            bikes = int(item['AvailableOrdinaryBikes'])+int(item['AvailableElectricBikes'])
             free = int(item['FreePlaces'])
             extra = {
                 'uid': item['Id'],
                 'slots': int(item['TotalPlaces']),
-                'address': re.sub(r'^\d+\s*-\s*', '', item['Address'])
+                'address': re.sub(r'^\d+\s*-\s*', '', item['Address']),
+                'ordinarySlots': int(item['TotalOrdinaryPlaces']),
+                'electricSlots': int(item['TotalElectricPlaces']),
+                'ordinaryBikes': int(item['AvailableOrdinaryBikes']),
+                'electricBikes': int(item['AvailableElectricBikes']),
```

Comment:
Since it seems this system differentiates the electric from the ordinary slots, would it be useful to also add the available slots on extra separated by type? like `electric_free` and `normal_free` mapped to `FreeElectricPlaces` and `FreeOrdinaryPlaces`

Reply:
OK, I didn't found any description for extra fields in `BikeShareStation` class, so I've used names from Velobike API.

I'll change the namings of extra fields and then push a commit.

Should be field `has_ebikes` used only when station has _available_ e-bikes, or it's used in any case when station support e-bikes docking? In last case we can check `StationType` array for property "electric".

Reply:
> Should be field has_ebikes used only when station has available e-bikes, or it's used in any case when station support e-bikes docking? In last case we can check StationType array for property "electric".

I think that would be the most correct for the property `has_ebikes`. I will push a fix on all other systems where possible to change this field as "this station supports ebikes". So yes, that would be for this system, if `StationType` contains `electric`. Thanks!

Reply:
As discussed, I have made the usage of `has_ebikes` consistent with this across systems: https://github.com/eskerda/pybikes/pull/410

Reply:
I've added `has_ebikes` in [7bcef0b commit](https://github.com/eskerda/pybikes/pull/409/commits/7bcef0b024d46a0a13797f240bd0618523818cb3). It will be set only if station supports ebikes docking, in other cases it will not be present in `extra` dictionary.